samples/auth-user-sample/authUserFlow.js

```
@@ -0,0 +1,156 @@
+async function exchangeCode(code) {
+  // Which should be downloaded from the Google Developers Console.
+  const oAuth2Client = new OAuth2Client(
+    keys.installed.client_id,
+    keys.installed.client_secret,
+    keys.installed.redirect_uris[0]
+  );
+
+  try {
+    const r = await oAuth2Client.getToken(code);
+    console.info(r.tokens);
+    return r.tokens;
+  } catch (err) {
+    console.log(err);
+  }
+}
+
+// [START bigquery_auth_user_flow]
+function authFlow(projectId = 'project_id') {
+  /**
+   * TODO(developer):
+   * Save Project ID as environment variable PROJECT_ID="project_id"
+   * Uncomment the following line before running the sample.
+   */
+  // const projectId = process.env.PROJECT_ID;
+
+  const credentials = {
+    type: 'authorized_user',
+    client_id: keys.installed.client_id,
+    client_secret: keys.installed.client_secret,
+    refresh_token: process.env.REFRESH_TOKEN,
+  };
+
+  return {
+    projectId,
+    credentials,
+  };
+}
+// [END bigquery_auth_user_flow]
+// [START bigquery_auth_user_query]
+async function query() {
+  const {BigQuery} = require('@google-cloud/bigquery');
+  const credentials = main.authFlow();
+  const bigquery = new BigQuery(credentials);
+  // Queries the U.S. given names dataset for the state of Texas.
+  const query = ` SELECT name, SUM(number) as total
+  FROM \`bigquery-public-data.usa_names.usa_1910_current\`
+  WHERE name = 'William'
+  GROUP BY name;`;
+
+  // For all options, see https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs/query
+  const options = {
+    query: query,
+  };
+
+  // Run the query as a job
+  try {
+    const [job] = await bigquery.createQueryJob(options);
+    console.log(`Job ${job.id} started.`);
+
+    // Wait for the query to finish
+    const [rows] = await job.getQueryResults();
+
+    // Print the results
+    console.log('Rows:');
+    rows.forEach(row => console.log(row));
+
+    return rows;
+  } catch (err) {
+    console.log(err);
```

No need for the try/catch here - just let it throw!